### ISSUE TITLE
null-Facts suppress rule-execution

### ISSUE THREAD

**Issue by pjotre86** [rulebook-rules/rulebook#107]
Hey there!
First of all: Just discovered RuleBook and I like the concept pretty much.
I'm just creating a small test-project to see what's possible and stumbled over a small uncovinient/unexpected behaviour.

I created some POJO-rules and want to execute them with
`rulebook.run(facts);`
The facts I create via:
```
NameValueReferableMap facts = new FactMap();
facts.setValue("oneFact",  "first");
facts.setValue("anotherFact", someObject);
```
Everything works fine so far, my rules are behaving like expected, but not when `someObject` is `null`.
Then nothing happens, no rule is triggered.
Is this intended behaviour?
I was expecting to be able to check facts to be null inside the rules themselves, but now I never get there.
Is there a different way to create the fact-map, so it works with null-values?
I helped myself now by sub-classing the `FactMap` like this:

```
private static class NullSafeFactMap extends FactMap {
    @Override
    public void setValue(String name, Object value) {
        if (value != null) {
            super.setValue(name, value);
        }
    }
}
```

That works for me, but I was wondering if I really need something like that.
Curious for your answers!

**Comment by pjotre86** [comment]
``

**Comment by Clayton7510** [comment]
It should work with null Facts. But if you are seeing something different then I'll check it out. It's possible this could be a bug. Thanks for raising this issue.

**Comment by pjotre86** [comment]
Thanks!
Do you need a more detailed example?

**Comment by Clayton7510** [comment]
If you have a complete example, that would be helpful. Thanks.

**Comment by pjotre86** [comment]
All right, here's qhat I got, I put it in a simple test-scenario:
Her the simple Rule I have. As you see it should always be executed:

```
package nullproblem;

import com.deliveredtechnologies.rulebook.RuleState;
import com.deliveredtechnologies.rulebook.annotation.*;

@Rule( name = "SimpleRule")
public class SimpleRule {
    @Given("status")
    private int status;
    @Given("object")
    private Object object;
    @Result
    private String result;

    @When
    public boolean when() {
        return true;
    }

    @Then
    public RuleState then() {
        result = "Rule was triggered with status=" + status + " and object=" + object;
        return  RuleState.NEXT;
    }
}
```

And here the corresponding tests:

```
package nullproblem;

import com.deliveredtechnologies.rulebook.FactMap;
import com.deliveredtechnologies.rulebook.NameValueReferableMap;
import com.deliveredtechnologies.rulebook.model.Auditor;
import com.deliveredtechnologies.rulebook.model.RuleStatus;
import com.deliveredtechnologies.rulebook.model.runner.RuleBookRunner;
import org.junit.Test;

import java.util.Map;

import static junit.framework.TestCase.assertEquals;

public class ShowNullProblemTest {
    @Test
    public void runWithNotNull() {
        RuleBookRunner ruleBook = new RuleBookRunner("nullproblem");
        NameValueReferableMap facts = new FactMap();
        facts.setValue("status", 1);
        facts.setValue("object", "A String");
        ruleBook.run(facts);
        Auditor auditor = (Auditor) ruleBook;
        Map<String, RuleStatus> auditMap = auditor.getRuleStatusMap();
        assertEquals(auditor.getRuleStatus("SimpleRule"), RuleStatus.EXECUTED);
        ruleBook.getResult().ifPresent(System.out::println);;
    }

    @Test
    public void runWithNull() {
        RuleBookRunner ruleBook = new RuleBookRunner("nullproblem");
        NameValueReferableMap facts = new FactMap();
        facts.setValue("status", 1);
        facts.setValue("object", null);
        ruleBook.run(facts);
        Auditor auditor = (Auditor) ruleBook;
        assertEquals(auditor.getRuleStatus("SimpleRule"), RuleStatus.EXECUTED);
        ruleBook.getResult().ifPresent(System.out::println);;
    }
}
```
When I execute the tests the "runWithNull"-tests fails, because the auditor says the rule was not executed, giving back RuleStatus.SKIPPED.
Here the result:
```
junit.framework.AssertionFailedError: 
Expected :SKIPPED
Actual   :EXECUTED
```

Just let me know if you need futher information.
Regards,
Peter

**Comment by Clayton7510** [comment]
Thanks for the detail. I see there is a problem with the facts filter in the rule implementation. It should be corrected by the end of this weekend.

Thanks for raising this!

**Comment by Clayton7510** [comment]
This is fixed in 0.9.1-SNAPSHOT

**Comment by pjotre86** [comment]
Cool, seems to work.
Thanks alot!
When will you release 0.9.1 ?

**Comment by Clayton7510** [comment]
This week. Possibly tonight.

**Comment by Clayton7510** [comment]
I had to go out to a client site this week and I did not have my deployment keys with me. So, I will be deploying tonight.

**Comment by Clayton7510** [comment]
v0.9.1 was released to maven central.

**Comment by pjotre86** [comment]
Cool, thanks alot! I will give it a try within the next couple of days.

**Comment by Clayton7510** [comment]
Closed due to fix released in v0.9.1.
